playbooks/azure.yml

```diff
@@ -0,0 +1,143 @@
+      "29":  "northeurope"
+      "30":  "westeurope"
+      "31":  "germanycentral"
+      "32":  "germanynortheast"
+      "33":  "ukwest"
+      "35":  "uksouth"
+
+  # These variable files are included so the azure-security-group role
+  # knows which ports to open
+  vars_files:
+    - roles/openconnect/defaults/main.yml
+    - roles/openvpn/defaults/main.yml
+    - roles/shadowsocks/defaults/main.yml
+    - roles/ssh/defaults/main.yml
+    - roles/streisand-gateway/defaults/main.yml
+    - roles/stunnel/defaults/main.yml
+    - roles/tor-bridge/defaults/main.yml
+    - roles/wireguard/defaults/main.yml
+
+  vars_prompt:
+    - name: "azure_region_var"
+      prompt: >
+        What region should the server be located in?
+           1. East US             (Virginia)
+           2. East US 2           (Virginia)
+           3. Central US          (Iowa)
+           4. North Central US    (Illinois)
+           5. South Central US    (Texas)
+           6. West Central US     (West Central US)
+           7. West US             (California)
+           8. West US 2           (West US 2)
+           9. US Gov Virginia     (Virginia)
+          10. US Gov Iowa         (Iowa)
+          11. US DoD East         (US DoD East)
+          12. US DoD Central      (US DoD Central)
+          13. Canada East         (Quebec City)
+          14. Canada Central      (Toronto)
+          15. Brazil South        (Sao Paulo State)
+          16. Southeast Asia      (Singapore)
+          17. East Asia           (Hong Kong)
+          18. Australia East      (New South Wales)
+          19. Australia Southeast (Victoria)
+          20. China East          (Shanghai)
+          21. China North         (Beijing)
+          22. Central India       (Pune)
+          23. West India          (Mumbai)
+          24. South India         (Chennai)
+          25. Japan East          (Tokyo, Saitama)
+          26. Japan West          (Osaka)
+          27. Korea Central       (Seoul)
+          28. Korea South         (Busan)
+          29. North Europe        (Ireland)
+          30. West Europe         (Netherlands)
+          31. Germany Central     (Frankfurt)
+          32. Germany Northeast   (Magdeburg)
+          33. UK West             (Cardiff)
+          35. UK South            (London)
+        Please choose the number of your region. Press enter for default (#9) region.
+      default: "28"
+      private: no
+
+    - name: "azure_instance_name_var"
+      prompt: "\nWhat should the server be named? Press enter for default (streisand).\n"
+      default: "streisand"
+      private: no
+
+    - name: "confirmation_credentials"
+      prompt: "\n\nEnsure that you have the azure credentials file: ~/.azure/credentials \nDetails on generating this can be found at \nhttps://docs.ansible.com/ansible/guide_azure.html#storing-in-a-file\nand \nhttps://docs.microsoft.com/en-in/azure/azure-resource-manager/resource-group-create-service-principal-portal\n"
```

Comment:
Unfortunately I felt the documentation located https://docs.ansible.com/ansible/guide_azure.html#storing-in-a-file and https://docs.microsoft.com/en-in/azure/azure-resource-manager/resource-group-create-service-principal-portal were lacking :(, Maybe in a different commit, it would be worth writing up better step by step (Maybe if someone uses Azure on a regular basis though the Ansible documentation is enough).